framework/src/utils/MatrixTools.C

```diff
@@ -31,7 +31,7 @@ int inverse(const std::vector<std::vector<Real> > & m, std::vector<std::vector<R
     for (const auto & matrix_entry : rowvec)
       A.push_back(matrix_entry);
 
-  int error = inverse(A, n);
+  int error = inverse(A, (int) n);
```

Comment:
i had a quick look on the internet, and it appears that your suggestion would cause no problems in all the C code.  I'm a little unsure about the fortran stuff - what if some version of fortran/architecture/compiler does something weird with signed and unsigned stuff?  It's tricky because i'll never know.  Anyway, i'll implement your suggestion and we'll see if you like it or not.